lib/zooniverse_user.rb

```diff
@@ -1,4 +1,77 @@
 class ZooniverseUser < ActiveRecord::Base
   establish_connection :"zooniverse_home_#{ Rails.env }"
   self.table_name = 'users'
+
+  attr_reader :password
+
+  def self.import_users(users=nil)
+    if users.blank?
+      self
+    else
+      where(login: users)
+    end.find_each(&:import)
+  end
+
+  def self.authenticate(username, password)
+    find_by(login: username).try(:authenticate, password)
+  end
+
+  def self.from_user(user)
+    zu = find_or_create_by(login: user.display_name) do |u|
```

Comment:
find_or_create_by with a save just below would be 2 persistence calls. Maybe just find_or_initialize_by and then save!